pageserver/src/tenant/mgr.rs

```diff
@@ -734,44 +785,58 @@ pub(crate) async fn upsert_location(
             }
 
             let new_slot = match &new_location_config.mode {
-                LocationMode::Secondary(_) => TenantSlot::Secondary,
+                LocationMode::Secondary(_) => {
+                    let tenant_path = conf.tenant_path(&tenant_id);
+                    unsafe_create_dir_all(&tenant_path)
```

Comment:
Should have comment explaining why unsafe is ok here.
(I don't understand why it's ok.)

Reply:
a3f728f278ae3a936c6a32825779a38e1825a0d0